datafusion/common/src/config.rs

```diff
@@ -2093,3 +2094,22 @@ mod tests {
         assert_eq!(parsed_metadata.get("key_dupe"), Some(&Some("B".into())));
     }
 }
+
+#[cfg(test)]
+mod tests_isolated {
+    // The point of this test is to check that the config_namespace! macro
+    // can compile without any surrounding `use` statements. Hence putting
+    // it into its own test module.
+    #[test]
+    fn check_config_namespace_macro() {
```

Comment:
Do you think it may be better to add this test/module in `lib.rs` 🤔 ?

Since the subtle addition of a `use super::*` in the future would effectively hide any regression here.

Reply:
I believe this location / test is exactly for ensuring macro's don't import things unnecessarily: https://github.com/apache/datafusion/tree/main/datafusion/core/tests/macro_hygiene

How about we put this test there?